### DISCUSSION TITLE
Writing new glb removes clear coat

### DISCUSSION THREAD

**Discussion by mpaleo** [donmccurdy/glTF-Transform#1156] [Q&A] [answered]
So I'm trying to optimize textures (ktx) from a glb exported from Blender. I'm using babylon as render engine.
For some reason when I read and write the new optimized glb, it removes clear coat values.
At first I thought it was an issue with `toktx` but then I removed that portion and still no clear coat values in babylon.
Is there any config value missing ?

```js
import {NodeIO} from '@gltf-transform/core'
import {toktx} from '@gltf-transform/cli'

export const optimizeGLB = async (buffer) => {
  const io = new NodeIO()

  const document = await io.readBinary(buffer)

  // await document.transform(
  //   toktx({ mode: 'etc1s', quality: 155 }),
  // )

  return io.writeBinary(document)
}
```

**Comment by donmccurdy** [comment, marked as answer]
I think there should be warnings in your console about unknown extensions, correct?

Extensions are not included in the /core module, they need to be imported and registered:

```js
import { NodeIO } from '@gltf-transform/core';
import { KHRONOS_EXTENSIONS } from '@gltf-transform/extensions';
const io = new NodeIO().registerExtensions(KHRONOS_EXTENSIONS);
```

https://gltf-transform.dev/extensions

If you're using non-Khronos extensions (EXT_meshopt_compression, EXT_mesh_gpu_instancing, EXT_texture_webp, ...) then you'd instead need to register `ALL_EXTENSIONS`.

**Comment by mpaleo** [comment]
Hey @donmccurdy
Yeah I was getting the following warnings before
```
Missing optional extension, "KHR_materials_emissive_strength".
Missing optional extension, "KHR_materials_clearcoat".
Missing optional extension, "KHR_materials_specular".
Missing optional extension, "KHR_materials_ior".
Missing optional extension, "KHR_materials_transmission".
Missing optional extension, "KHR_materials_emissive_strength".
```
I tried registering `KHRONOS_EXTENSIONS` and `ALL_EXTENSIONS` a few mins ago but still no clear coat

**Reply by donmccurdy** [reply]
Registering the extensions before reading is all that's required -- if that's not working please share a way to reproduce the issue, thanks!
